packages/keyboard-shortcuts/src/hooks/use-shortcut.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import Mousetrap from 'mousetrap';
+import 'mousetrap/plugins/global-bind/mousetrap-global-bind';
+import { includes } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { rawShortcut } from '@wordpress/keycodes';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Return true if platform is MacOS.
+ *
+ * @param {Object} _window   window object by default; used for DI testing.
+ *
+ * @return {boolean}         True if MacOS; false otherwise.
+ */
+function isAppleOS( _window = window ) {
+	const { platform } = _window.navigator;
+
+	return platform.indexOf( 'Mac' ) !== -1 ||
+		includes( [ 'iPad', 'iPhone' ], platform );
+}
+
+/**
+ * Attach a keyboard shortcut handler.
+ *
+ * @param {string} name       Shortcut name.
+ * @param {Function} callback Shortcut callback.
+ * @param {Object} options    Shortcut options.
+ */
+function useShortcut( name, callback, {
+	bindGlobal = false,
+	eventName = 'keydown',
+	target,
+} ) {
+	const { combination, aliases } = useSelect( ( select ) => {
+		return {
+			combination: select( 'core/keyboard-shortcuts' ).getShortcutKeysCombination( name ),
+			aliases: select( 'core/keyboard-shortcuts' ).getShortcutAliases( name ),
+		};
+	}, [ name ] );
+
+	useEffect( () => {
+		const shortcuts = [ combination, ...aliases ].filter( ( shortcut ) => !! shortcut );
+		const mousetrap = new Mousetrap( target ? target.current : document );
```

Comment:
Where is `target` documented? Is it supposed to be a `useRef` value? If we use this with a ref, will there be an initial (unintended) binding to the document before it's bound to the target?

Reply:
It's a ref yes and there won't be unintended binding, this is a common practice for several hooks in the community that rely on refs, because we're using `useEffect` the ref will already be initialized.